### ISSUE TITLE
Improve tab support on Pods field-list-item

### ISSUE THREAD

**Issue by heybran** [pods-framework/pods#7196]
## 1. `pods-field_name` does not have tab focus indicator/outline.

The `compan_name` here is currently being focused by Tab, but we're using a `div` as a `button`, it loses the default browser outline so users have no ideas where they're currently focused on.

https://github.com/pods-framework/pods/blob/main/ui/js/dfv/src/admin/edit-pod/main-tabs/field-list-item.js#L247
```
				<div
					tabIndex={ 0 }
					role="button"
					className="pods-field pods-field_name"
					onClick={ onEditFieldClick }
					onKeyPress={ handleKeyPress }
				>
					{ name }
				</div>
```

<img width="1293" alt="Screenshot 2023-10-03 at 11 25 31" src="https://github.com/pods-framework/pods/assets/75633537/d61d1f84-7be1-4442-9131-82737f16aaa4">

## 2. `Edit`/`Duplicate`/`Delete` buttons 

Similar to first issue, these buttons should be visible when they're tabbed into, on the screenshot they're visible as I'm currently hovering on the field wrapper with my mouse.

<img width="1293" alt="Screenshot 2023-10-03 at 11 32 47" src="https://github.com/pods-framework/pods/assets/75633537/75b36983-d705-4410-8d35-5de69de3a19e">

### Proposed Solution

1. Replace `div` with `button` but style it as text button or add focus outline to `div[tabindex="0"]`

2. Add a `focus:within` to `pods-field_outer-wrapper` to show `Edit`/`Duplicate`/`Delete` buttons.

**Comment by heybran** [comment]
After fixes, all buttons are focus visible to users as they navigate with keyboards . Will play with UI a bit more and see if there are more areas that need improvement.

https://github.com/pods-framework/pods/assets/75633537/f5eca59f-6792-4833-8dbe-3ca62b6c0784

**Comment by heybran** [comment]
Sorry, just noticed my commit ended with a huge git diff on this file `ui/js/dfv/src/admin/edit-pod/main-tabs/field-list-item.js` https://github.com/heybran/pods/commit/b18c5b81e028cf95aba226c1306c30aaf000500a, it was caused by my VS Code indent setting. 

Please see this comment https://github.com/heybran/pods/commit/b18c5b81e028cf95aba226c1306c30aaf000500a#r128990258

**Comment by heybran** [comment]
Sorry, just noticed my commit ended with a huge git diff on this file `ui/js/dfv/src/admin/edit-pod/main-tabs/field-list-item.js` https://github.com/heybran/pods/commit/b18c5b81e028cf95aba226c1306c30aaf000500a, it was caused by my VS Code indent setting. 

Please see this comment https://github.com/heybran/pods/commit/b18c5b81e028cf95aba226c1306c30aaf000500a#r128990258

**Comment by heybran** [comment]
Hi @sc0ttkclark, I think I can close this issue as PR for this one has been merged and released.
